share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```
@@ -380,5 +380,58 @@ function(sysbuild_module_call)
       endforeach()
     endif()
   endforeach()
+endfunction()
+
+# Usage:
+#   sysbuild_cache_set(VAR <variable> [APPEND [LIST]] <value>)
+#
+# This function will set the specified value of the sysbuild cache variable in
+# the CMakeCache.txt file which can then be accessed by images.
+# `VAR` specifies the variable name to set/update.
+#
+# The result will be returned in `<variable>`.
+#
+# Example use:
+#   sysbuild_cache_set(VAR ATTRIBUTES APPEND LIST battery)
+#     Will add the item `battery` to the `ATTRIBUTES` variable as a new element
+#     in the list in the CMakeCache.
+#
+# <variable>: Name of variable in CMake cache.
+# APPEND:     If specified then will append the supplied data to the existing
+#             value (unless LIST is also specified, this will be treated as a
+#             string, no space character is added between appends).
+# LIST:       Will treat the append item as a list and add the new data as a
+#             new element to the list.
+# <value>:    Value to set/update.
+function(sysbuild_cache_set)
+  cmake_parse_arguments(VARS "APPEND;LIST" "VAR" "" ${ARGN})
+
+  if(NOT VARS_UNPARSED_ARGUMENTS AND VARS_APPEND)
+    message(FATAL_ERROR "sysbuild_set(VAR <var> APPEND ...) missing required arguments to append")
+  elseif(VARS_LIST AND NOT VARS_APPEND)
+    message(FATAL_ERROR "sysbuild_set(VAR <var> APPEND LIST ...) is required for the LIST option")
+  elseif(NOT VARS_VAR)
+    message(FATAL_ERROR "sysbuild_set(VAR <var> ...) missing required VAR variable")
+  endif()
```

we now have helper macros in place to simplify testing and ensure some consistency in error messaging, so let's start to use them 😉 :
```suggestion
zephyr_check_arguments_required("sysbuild_set" VARS VAR)
```

---

Fixed